### ISSUE TITLE
Clicking anywhere outside a modal dialog should close it

### ISSUE THREAD

**Issue by imagiMary** [status-im/status-desktop#9702]
Version: 0.9
OS: Windows 10

Clicking anywhere outside a modal dialogue should close it. This functionality currently doesn’t work for the modal dialogs listed below. 

***

<ins>Steps to reproduce:</ins>

1. Navigate to:

- Setting/Profile/Change Password
- Setting/Keycard/Setup a new Keycard with an existing account
- Setting/Keycard/Create a new Keycard account with a new seed phrase
- Setting/Keycard/Import or restore via a seed phrase
- Setting/Keycard/Import from Keycard to Status Desktop
- Setting/Keycard/Check what’s on a Keycard
-  Setting/Keycard/Factory reset a Keycard
- Wallet/Add account

<ins>What currently incorrectly happens:</ins>

If the user clicks anywhere outside of the modal dialog, nothing happens.  

<ins>What is supposed to happen:</ins>

If the user clicks anywhere outside of the modal dialog, the modal dialogue should close.

<ins>Other information</ins>

https://user-images.githubusercontent.com/124154641/222198332-429d0614-bde4-4ab9-9204-a451e3a12c3d.mp4

**Comment by caybro** [comment]
These kind of dialogs (password/keycard/...) are not closable on purpose; there must be some intention behind this.

**Comment by imagiMary** [comment]
@caybro I don't think there is any intention behind it, nobody in the design team could think of any intention behind this because the user can always click the 'x' to close the dialog at any time and clicking outside the dialog should be equivalent to clicking the 'x' close button in the dialog window.

**Comment by caybro** [comment]
CC @saledjenic

**Comment by saledjenic** [comment]
@imagiMary yes, all keycard flows share the same closing policy logic which is set that user is able to close a popup only interacting with buttons displayed in the footer of that popup or upper right "X" (close) button if it is displayed. If "X" button is displayed user is able to close the popup pressing `Esc` key as well, otherwise only using buttons from the footer section.

If there is a request for the change we can change that, but we will still have some states in all those flows where we will disable any user interaction with the popup or enable only the one we want user to interact with.

For example when user runs the change password flow, the last step of that flow, if everything went well, will be a popup which cannot be closed anyhow (except if user close the app) and the only button which will be enabled is "Restart app" button, cause in such case we have to force user to restart the app cause the db is re-encypted and we cannot let user continue using it.

Also for example if user runs "Setup a new Keycard with an existing account" flow and migrating a profile keypair, in this screen the migrations is happening and notice that everything is disabled, so we don't allow user to do anything while that is ongoing:

![a1](https://user-images.githubusercontent.com/86303051/222707869-5c796506-3de5-49b5-903a-a510095bc692.png)

Then once migration is completed, we offer user only a single button enabled, so the only user actually can do at this step is to restart the app (no close popup, no anything else):

![a2](https://user-images.githubusercontent.com/86303051/222708023-d973cb3b-f8df-49bc-ab7b-07186f29af56.png)

**Comment by imagiMary** [comment]
Good point, thanks for your reply 🙂 

Given the usecase that you have correctly pointed out, lets amend the change request to the following; 

"Clicking anywhere outside a modal dialogue which has a close 'x' **visible** in the top right corner, should close the dialogue.  e.g. if there is a close 'x' visible, clicking outside of the modal dialogue should be equivalent to clicking the close 'x'".

Modal dialogues that currently have a close 'x', but where clicking outside doesn't close them (but should if the 'x' is visible):

- Change password dialogue where the user enters their new password (as the user should be able to exit by closing the dialogue before they click the "Change password and restart Status" button.

- Wallet 'Add account' dialogue.

- All 'keycard dialogues' listed in the original issue report (but only when a close 'x' is visible in the dialogue of course).

To summarize, if a user isn't able to close a dialogue by clicking on an 'x' then clicking outside the dialogue should do nothing, but if a user is able to close the dialogue by clicking on an 'x', then clicking outside the dialogue (as an alternative to clicking the 'X) should close the dialogue.

**Comment by saledjenic** [comment]
@imagiMary I agree with all you've said. Assigning this to myself and will sort that out once I am done with some other, higher priority issues.
